tools/src/main/java/com/nedap/archie/archetypevalidator/validations/SpecializedDefinitionValidation.java

```diff
@@ -155,6 +162,15 @@ private void validateConformsTo(CObject cObject, CObject parentCObject) {
 
     }
 
+    private boolean hasConformingParent(CAttribute parentAttribute, CPrimitiveObject member) {
+        for(CObject parentCObject:parentAttribute.getChildren()) {
+            if(member.cConformsTo(parentCObject, (a, b) -> combinedModels.rmTypesConformant((String) a, (String) b))) {//TODO: why is casting necessary for compiler?
```

Comment:
> why is casting necessary for compiler?

Because you have a "Raw use of parameterized class 'CPrimitiveObject'" in the method declaration.

```suggestion
    private boolean hasConformingParent(CAttribute parentAttribute, CPrimitiveObject<?, ?> member) {
        for(CObject parentCObject:parentAttribute.getChildren()) {
            if(member.cConformsTo(parentCObject, (a, b) -> combinedModels.rmTypesConformant(a, b))) {
```

Reply:
thanks